### ISSUE TITLE
Update ssl code

### ISSUE THREAD

**Issue by static-void** [ColumPaget/Movgrab#3]
I just built this. I cannot get it to build with ssl support. It builds successfully without, but cannot download from youtube (lists no formats available). It also seg faults with the -t option. Would love it if this was working again!

**Comment by static-void** [comment]
Looking a bit more closely, it seems youtube requires ssl support. However, it appears movgrab does not want to link with ssl enabled.

**Comment by ColumPaget** [comment]
I'll try to look into this this weekend. Movgrab is pretty old, and needs a major wash and brush up.

Can I ask what ssl library you're using? I assume it's open ssl? What error gets printed when you try to build with ssl support?

**Comment by static-void** [comment]
Yes, openssl on ubuntu 18.04 (x64). If I use libssl1.0-dev and libssl1.0.0 then ssl is detected ok by configure (with --enable-ssl) but I get a lot of undefined references. I tried playing with the linker flags a bit to explicitly link to the correct library, but it made no difference. If I install libssl-dev and libssl1.1 instead (the latest openssl), then configure fails to find libssl at all. I also noticed that there are much newer versions of libuseful so I tried building the latest and linking movgrab to that, but that was not an easy fix as I guess the libuseful API has changed.

Would it be better to just try linking against e.g. LibTomCrypt? Is that supported?

This is a really great tool and it would be awesome if it were working again. There are many low power machines that really benefit from a video scraping tool that does not rely on python (e.g. the zipit z2), and in general, why crack a nut with a sledgehammer anyway? :D

**Comment by ColumPaget** [comment]
Yes, I'm afraid movgrab uses an ancient version of libUseful. I'm hoping to get some time this weekend to upgrade it to use the latest version. Can I ask, did libUseful compile against openssl okay? I use libressl myself, and I always assumed that if a thing compiles against that, it will compile against openssl too, but maybe that's not the case?

**Comment by static-void** [comment]
Yes, the latest from the master branch of https://github.com/ColumPaget/libUseful links fine against libssl1.0.0 when configured with --enable-ssl:
```
ed@fermat01:~/project/libUseful$ ldd libUseful-4.5.so 
	linux-vdso.so.1 (0x00007ffce6bd2000)
	libcrypto.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007f8c679d5000)
	libssl.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007f8c6776d000)
        ...
```

**Comment by ColumPaget** [comment]
Great! Then that's the way forwards. Hopefully I'll have it ready next week, or the week after.

**Comment by static-void** [comment]
Hi Colum, did you have any luck with this?

**Comment by ColumPaget** [comment]
Yes, I finally found some time to work on it. I've not made a release yet, because I'm still making changes, but if you clone the repo you should get a version that builds with ssl on more recent systems.

**Comment by static-void** [comment]
This is awesome news, thanks a lot, very much appreciated. I will give it a try now.

**Comment by ColumPaget** [comment]
Let me know how it goes, travis-ci.com says it builds okay, but you never know.

**Comment by static-void** [comment]
It built and works perfectly, though tested only on youtube for now

**Comment by ColumPaget** [comment]
very few of the sites work, that's the main work I've got to do for the next release. Over time sites go out of business, or change the way they deliver content. It's hard to stay on top of!

**Comment by sxiii** [comment]
@ColumPaget I've just tried to build Movgrab on ArchLinux and it doesen't see the flag --enable-ssl either. Not enabling SSL support. Are there any updates on that?

**Comment by ColumPaget** [comment]
hi sxii,

where are you getting the source from? I've just checked and compiling with --enable-ssl works fine on the master branch?
